ui/src/main/java/com/dat3m/ui/options/OptionsPane.java

```diff
@@ -179,9 +218,321 @@ private void mkGrid() {
+        for (Map.Entry<String, String> entry : extraOptionsMap.entrySet()) {
+            text.append(init ? " --" : "--").append(entry.getKey()).append('=').append(entry.getValue());
+            init = true;
+        }
+        extraOptionsField.setText(text.toString());
+    }
+
+    private void handleExtraOptionsButton(ActionEvent e) {
+        updateExtraOptionsFields();
+        extraOptionsDialog.setVisible(true);
+        extraOptionsDialog.setLocationRelativeTo(this);
+        extraOptionsDialog.requestFocus();
+    }
+
+    private void doExport() {
+        if (extraOptionsDialog.isVisible()) {
+            toText();
+            extraOptionsDialog.setVisible(false);
+        }
+        configurationFileChooser.showSaveDialog(this);
+        final File file = configurationFileChooser.getSelectedFile();
+        if (file == null) {
+            return;
+        }
+        try {
+            final Configuration properties = Configuration.builder()
+                    .setOptions(extraOptionsMap)
+                    .setOption(METHOD, methodPane.getSelectedItem().name())
+                    .setOption(SOLVER, solverPane.getSelectedItem().name())
+                    .setOption(PROPERTY, propertyPane.getSelectedItem().name())
+                    .setOption(TARGET, targetPane.getSelectedItem().name())
+                    .setOption(PROGRESSMODEL, progressPane.getSelectedItem().name())
+                    .setOption(BOUND, boundField.getText())
+                    .setOption(TIMEOUT, timeoutField.getText())
+                    .setOption(WITNESS, Boolean.toString(showViolationField.isSelected()))
+                    .build();
+            //NOTE the properties file format almost fits the format accepted by Configuration.loadCharSource.
+            //But comments missing a whitespace after '#' are treated as directives.
+            try (FileWriter writer = new FileWriter(file)) {
+                writer.append("# Created with Dartagnan\n# ")
+                        .append(new Date().toString())
+                        .append('\n')
+                        .append(properties.asPropertiesString());
+            }
+        } catch (IOException | InvalidConfigurationException e) {
+            showError(e.getMessage(), "Error while exporting configuration");
+        }
+    }
+
+    private void doImport() {
+        if (extraOptionsDialog.isVisible()) {
+            extraOptionsDialog.setVisible(false);
+        }
+        configurationFileChooser.showOpenDialog(this);
+        final File file = configurationFileChooser.getSelectedFile();
+        if (file == null) {
+            return;
+        }
+        final var properties = new HashMap<String, String>();
+        try {
+            final Configuration config = Configuration.builder().loadFromFile(file.toPath()).build();
+            for (String key : List.copyOf(config.getUnusedProperties())) {
+                properties.put(key, config.getProperty(key));
```

Comment:
By using `Configuration` for importing the file, the UI also supports `#include` directives in the format, just like `Dartagnan`.

The method `getProperty` is deprecated, though.  The best alternative I have:
```java
Properties props = new Properties();
props.load(new StringReader(config.asPropertiesString()));
```

Reply:
Just use `getProperty()` even if it is deprecated.